src/coreclr/jit/gentree.cpp

```diff
@@ -30360,8 +30360,11 @@ GenTree* Compiler::gtFoldExprHWIntrinsic(GenTreeHWIntrinsic* tree)
 #endif // !TARGET_XARCH && !TARGET_ARM64
 
                     DEBUG_DESTROY_NODE(op, tree);
-                    INDEBUG(vectorNode->gtDebugFlags |= GTF_DEBUG_NODE_MORPHED);
 
+                    if (fgGlobalMorph)
+                    {
+                        INDEBUG(vectorNode->gtDebugFlags |= GTF_DEBUG_NODE_MORPHED);
+                    }
```

Comment:
The other `INDEBUG(vectorNode->gtDebugFlags |= GTF_DEBUG_NODE_MORPHED);` are properly under `if (fgGlobalMorph)` already

Reply:
generally, it's enough to call `fgMorphTreeDone(vectorNode)` in these `gtFoldExpr*` apis - it will take care about `fgGlobalMorph` and the `GTF_DEBUG_NODE_MORPHED` flag (it also clears local assertionprop info, but that is important only if your function is called from morph and on an existing tree).